ils_middleware/dags/penn.py

```diff
@@ -0,0 +1,179 @@
+from datetime import datetime, timedelta
+
+from airflow import DAG
+from airflow.models import Variable
+from airflow.operators.dummy import DummyOperator
+from airflow.operators.python import PythonOperator
+from airflow.utils.task_group import TaskGroup
+
+from ils_middleware.tasks.amazon.s3 import get_from_s3, send_to_s3
+from ils_middleware.tasks.amazon.sqs import SubscribeOperator, parse_messages
+from ils_middleware.tasks.sinopia.local_metadata import new_local_admin_metadata
+from ils_middleware.tasks.sinopia.email import (
+    notify_and_log,
+    send_update_success_emails,
+)
+from ils_middleware.tasks.sinopia.login import sinopia_login
+from ils_middleware.tasks.sinopia.metadata_check import existing_metadata_check
+from ils_middleware.tasks.sinopia.rdf2marc import Rdf2Marc
+from ils_middleware.tasks.alma.new import NewMARCtoAlma
+from ils_middleware.tasks.alma.mod_xml import to_alma_xml
+from ils_middleware.tasks.alma.match import match_marc_in_alma
+
+
+def task_failure_callback(ctx_dict) -> None:
+    notify_and_log("Error executing task", ctx_dict)
+
+
+def dag_failure_callback(ctx_dict) -> None:
+    notify_and_log("Error executing DAG", ctx_dict)
+
+
+default_args = {
+    "owner": "airflow",
+    "depends_on_past": False,
+    "email": ["airflow@example.com"],
+    "email_on_failure": False,
+    "email_on_retry": False,
+    "retries": 1,
+    "retry_delay": timedelta(minutes=5),
+    "provider": None,
+    "provide_context": True,
+    "on_failure_callback": task_failure_callback,
+}
+
+with DAG(
+    "penn",
+    default_args=default_args,
+    description="Penn Alma DAG",
+    schedule_interval=timedelta(minutes=5),
+    start_date=datetime(2021, 8, 24),
+    tags="alma",
```

Comment:
The DAG object is expecting a list of tags, to correct just change to `tags=["alma"],`